lib/trino-hive-formats/src/main/java/io/trino/hive/formats/line/grok/Match.java

```diff
@@ -0,0 +1,291 @@
+        }
+        capture.clear();
+
+        Map<String, String> mappedw = GrokUtils.namedGroups(this.match, this.subject);
+        Iterator<Entry<String, String>> it = mappedw.entrySet().iterator();
+        while (it.hasNext()) {
+            @SuppressWarnings("rawtypes")
+            Map.Entry pairs = (Map.Entry) it.next();
+            String key = null;
+            Object value = null;
+            if (this.grok.getNamedRegexCollectionById(pairs.getKey().toString()) == null) {
+                key = pairs.getKey().toString();
+            }
+            else if (!this.grok.getNamedRegexCollectionById(pairs.getKey().toString()).isEmpty()) {
+                key = this.grok.getNamedRegexCollectionById(pairs.getKey().toString());
+            }
+            if (pairs.getValue() != null) {
+                value = pairs.getValue().toString();
+
+                ImmutableMap<String, Object> keyValue = Converter.convert(key, value, grok);
+                key = keyValue.keySet().iterator().next();
+
+                // resolve value
+                if (keyValue.get(key) instanceof String) {
+                    value = cleanString((String) keyValue.get(key));
+                }
+                else {
+                    value = keyValue.get(key);
+                }
+            }
+            if (value != null) {
+                capture.put(key, value);
+            }
+            it.remove(); // avoids a ConcurrentModificationException
+        }
+    }
+
+    /**
+     * remove from the string the quote and double quote.
+     *
+     * @param value string to pure: "my/text"
+     * @return unquoted string: my/text
+     */
+    private String cleanString(String value)
+    {
+        if (value == null) {
+            return null;
+        }
+        if (value.isEmpty()) {
+            return value;
+        }
+        char[] tmp = value.toCharArray();
+        if (tmp.length == 1 && (tmp[0] == '"' || tmp[0] == '\'')) {
+            value = ""; //empty string
+        }
+        else if ((tmp[0] == '"' && tmp[value.length() - 1] == '"')
+                || (tmp[0] == '\'' && tmp[value.length() - 1] == '\'')) {
+            value = value.substring(1, value.length() - 1);
+        }
+        return value;
+    }
+
+    /**
+     * Get the json representation of the matched element.
+     * <p>
+     * example: map [ {IP: 127.0.0.1}, {status:200}] will return {"IP":"127.0.0.1", "status":200}
+     * </p>
+     * If pretty is set to true, json will return prettyprint json string.
+     *
+     * @return Json of the matched element in the text
+     */
+    public String toJson(Boolean pretty)
+            throws JsonProcessingException
+    {
+        if (capture == null) {
+            return "{}";
+        }
+        if (capture.isEmpty()) {
+            return "{}";
+        }
+
+        this.cleanMap();
+        if (pretty) {
+            objectMapper.enable(SerializationFeature.INDENT_OUTPUT);
```

Comment:
This ends up not being threadsafe, because `ObjectMapper#enable` will mutate the configuration on the mapper which will stay set on all subsequent calls. You probably want something along these lines instead:
```
return objectMapper.writerWithDefaultPrettyPrinter().writeValueAsString(capture);
```